interfaces/builtin/remoteproc.go

```diff
@@ -0,0 +1,61 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const remoteprocSummary = `allows access to remoteproc framework`
+
+const remoteprocBaseDeclarationPlugs = `
+  remoteproc:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const remoteprocBaseDeclarationSlots = `
+  remoteproc:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const remoteprocConnectedPlugAppArmor = `
+# Description: Remote Proc sysfs interface.
+
+/sys/devices/platform/**/remoteproc/remoteproc[0-9]/firmware rw,
+/sys/devices/platform/**/remoteproc/remoteproc[0-9]/state rw,
+/sys/devices/platform/**/remoteproc/remoteproc[0-9]/name r,
+/sys/devices/platform/**/remoteproc/remoteproc[0-9]/coredump rw,
+/sys/devices/platform/**/remoteproc/remoteproc[0-9]/recovery rw,
+
+# Allow adding new path for firmware location.
+/sys/module/firmware_class/parameters/path rw,
```

Comment:
How is this related to remote processors?

Reply:
The default firmware path is `/lib/firmware` and managed by kernel snap. As an application developer, if you want your kernel to look for external firmware is a different path, you would need to pass this path to `/sys/module/firmware_class/parameters/path`. So, this allows developers to load a firmware from the path they have decided.

Reply:
Correct, this is why I think this interface must be super-privilege.

Reply:
I still do not think that this belongs here, but to a complementary hopefully preexisting interface, as the scope of this - super-privileged or not - is much larger that what the interface might make one think. I'm interested in security feedback on this

Reply:
I agree - this feels far to broad / non-specific for this current interface interface.

Reply:
As discussed previously I think this could go under kernel-module-control or perhaps a new kernel-firmware-control ?

cc @alexmurray

Reply:
I think a separate interface since I would rather not grant full kernel-module-control when it is not needed.